filebeat/registrar/migrate.go

```diff
@@ -26,15 +26,51 @@ import (
 
 	"github.com/pkg/errors"
 
+	"github.com/elastic/beats/v7/filebeat/config"
+	"github.com/elastic/beats/v7/filebeat/input/file"
 	helper "github.com/elastic/beats/v7/libbeat/common/file"
 	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/paths"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/beats/v7/libbeat/statestore/backend/memlog"
 )
 
+type registryVersion string
+
 const (
-	legacyVersion  = "<legacy>"
-	currentVersion = "0"
+	noRegistry    registryVersion = ""
+	legacyVersion                 = "<legacy>"
+	version0                      = "0"
+	version1                      = "1"
 )
 
+const currentVersion = version1
+
+type Migrator struct {
+	dataPath    string
+	migrateFile string
+	permissions os.FileMode
+}
+
+func NewMigrator(cfg config.Registry) *Migrator {
+	path := paths.Resolve(paths.Data, cfg.Path)
+	migrateFile := cfg.MigrateFile
+	if migrateFile != "" {
+		migrateFile = paths.Resolve(paths.Data, migrateFile)
+	}
+
+	return &Migrator{
+		dataPath:    path,
+		migrateFile: migrateFile,
+		permissions: cfg.Permissions,
+	}
+}
+
+func (m *Migrator) Run() error {
+	return ensureCurrent(m.dataPath, m.migrateFile, m.permissions)
+}
+
+// ensureCurrent migrates old registry versions to the most recent version.
```

Comment:
`ensureCurrent`, `migrateLegacy` and `migrateVersion1` could be functions of `Migrator`. Thus, you would not need to pass around their parameters. Also, the names (and bodies) of these functions imply they are responsible for migrating the registry.